File: frontend/lib/models/appointment.dart
class Appointment {
  final int? id;
  final int patientId;
  final int doctorId;
  final String appointmentDate;
  final String reason;
  final String status;

  Appointment({
    this.id,
    required this.patientId,
    required this.doctorId,
    required this.appointmentDate,
    required this.reason,
    this.status = "Pending",
  });

  factory Appointment.fromJson(Map<String, dynamic> json) {
    return Appointment(
      id: json["id"],
      patientId: json["patient_id"],
      doctorId: json["doctor_id"],
      appointmentDate: json["appointment_date"],
      reason: json["reason"],
      status: json["status"] ?? "Pending",
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "patient_id": patientId,
      "doctor_id": doctorId,
      "appointment_date": appointmentDate,
      "reason": reason,
    };
  }
}

File: backend/app/models/appointment.py
from sqlalchemy import Column
from sqlalchemy import DateTime
from sqlalchemy import ForeignKey
from sqlalchemy import Integer
from sqlalchemy import String
from sqlalchemy.sql import func

from app.database import Base


class Appointment(Base):
    __tablename__ = "appointments"

    id = Column(
        Integer,
        primary_key=True,
        index=True,
    )

    patient_id = Column(
        Integer,
        ForeignKey("patients.id"),
        nullable=False,
    )

    doctor_id = Column(
        Integer,
        ForeignKey("users.id"),
        nullable=False,
    )

    appointment_date = Column(
        DateTime,
        nullable=False,
    )

    reason = Column(
        String,
        nullable=False,
    )

    status = Column(
        String,
        default="Pending",
    )

    created_at = Column(
        DateTime,
        server_default=func.now(),
    )